docs/source/networks.rst

```diff
@@ -0,0 +1,311 @@
+Networks
+========
+
+Depending on an experiment's objectives, there are different ways that
+experiment participants can interact with each other and with the
+experiment's stimuli. For some experiments, participants may receive the
+same initial stimuli and process it individually. For other experiments,
+they may sequentially interact with the stimuli. Some experiments may
+require participants to interact among themselves in various ways.
+
+In Dallinger, these interactions among participants and stimuli are
+represented using networks. Each participant and each stimulus represent a
+node in a network. The way these nodes are connected to each other is known
+as a `network topology`. For brevity, we will use the term `network` from now
+on when discussing Dallinger network topologies.
+
+Dallinger comes with a variety of networks that can be used by experimenters,
+and it's possible both to extend these networks or create completely new ones
+as well. The networks included in Dallinger are:
+
+- Empty.
+- Chain.
+- DelayedChain.
+- Star.
+- Burst.
+- FullyConnected.
+- DiscreteGenerational.
+- ScaleFree.
+- SequentialMicrosociety.
+- SplitSampleNetwork.
+
+Nodes and Sources
+-----------------
+
+In these networks, each participant is considered as a node. There is also a
+special kind of node, known as a `source`, which transmits information to
+other nodes. Sources are used in Dallinger as a means to send the stimuli to
+the participants. Not all experiments have sources, though. A chatroom
+experiment, for example, could just rely on user interactions and not
+require any other stimuli.
+
+All nodes have methods named `transmit` and `receive`, for sending and
+receiving information to or from other nodes. These methods can be used when
+adding a node to allow any specialized communication between nodes that an
+experiment may require.
+
+Nodes can have a `fitness` property, which is a number that can be used in
+some network models. The basic networks do not use this property.
+
+Some networks require that nodes have other properties, so for properly using
+those networks, an experiment would need to add these properties to its
+nodes.
+
+Node connections
+----------------
+
+A node can be connected to another node in three ways. A single direction
+connection to another node (to), a single direction connection from another
+node (from), or a bidirectional connection to another node (both). A node can
+transmit information when connected `to` another node. It can receive
+information when connected `from` another node. If it is connected to another
+node in `both` directions, it can both receive and transmit.
+
+Note that sources can only transmit information, so the only connection type
+allowed for a source node is `to`.
```

Comment:
I think this bit is slightly unclear. It's possible to misread it as node to source rather than source to node. Maybe a single line example?